tools/cloud-build/provision/daily-tests.tf

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_cloudbuild_trigger" "daily_test" {
+  for_each    = data.external.list_tests_py.result
+  name        = "DAILY-test-${each.key}"
+  description = "Runs the '${each.key}' integration test against `develop`"
+  tags        = [local.notify_chat_tag]
+
+  git_file_source {
+    path      = "tools/cloud-build/daily-tests/builds/${each.key}.yaml"
+    revision  = local.ref_develop
+    uri       = var.repo_uri
+    repo_type = "GITHUB"
+  }
+
+  source_to_build {
+    uri       = var.repo_uri
+    ref       = local.ref_develop
+    repo_type = "GITHUB"
+  }
+}
+
+module "daily_test_schedule" {
+  source   = "./trigger-schedule"
+  for_each = data.external.list_tests_py.result
+  trigger  = google_cloudbuild_trigger.daily_test[each.key]
```

Comment:
Should this not match the name name of the above resource? ("DAILY-test-${each.key}")

Reply:
`trigger` is not a name (string) but a reference to resource,
`google_cloudbuild_trigger.daily_test` is a collection (map) of resources, elements are referenced by `[]`